kvbc/include/merkle_builder/BaseMerkleBuilder.h

```diff
@@ -0,0 +1,76 @@
+// Concord
+//
+// Copyright (c) 2023 VMware, Inc. All Rights Reserved.
+//
+// This product is licensed to you under the Apache 2.0 license (the "License").
+// You may not use this product except in compliance with the Apache 2.0 License.
+//
+// This product may include a number of subcomponents with separate copyright
+// notices and license terms. Your use of these subcomponents is subject to the
+// terms and conditions of the sub-component's license, as noted in the
+// LICENSE file.
+
+#pragma once
+
+#include "IMerkleBuilder.h"
+#include "kvbc_app_filter/kvbc_key_types.h"
+#include "categorization/base_types.h"
+
+namespace concord {
+namespace kvbc {
+namespace sparse_merkle {
+
+using Hasher = concord::kvbc::categorization::Hasher;
+
+class BaseMerkleBuilder : public IMerkleBuilder {
+ public:
+  bool needProcessing(char type) const {
+    if (type == kKvbKeyEthBalance || type == kKvbKeyEthCode || type == kKvbKeyEthStorage || type == kKvbKeyEthNonce) {
```

Comment:
You assume here that it's an Ethereum deployment, can we do it without this assumption? maybe have the set of chars/strings to save as input?

Reply:
The general low level part of the MPT is in [sparse_merkle](https://github.com/vmware/concord-bft/tree/master/kvbc/include/sparse_merkle).